src/main/host/syscall/socket.c

```diff
@@ -580,7 +598,12 @@ SysCallReturn _syscallhandler_sendtoHelper(SysCallHandler* sys, int sockfd,
         /* TODO: Dynamically compute size based on how much data is actually
          * available in the descriptor. */
         size_t sizeNeeded = MIN(bufSize, SYSCALL_IO_BUFSIZE);
-        const void* buf = process_getReadablePtr(sys->process, sys->thread, bufPtr, sizeNeeded);
+
+        const void* buf = NULL;
+        /* If the size is 0, nothing should be trying to read the memory so NULL should be okay. */
+        if (sizeNeeded > 0) {
```

Comment:
It worked for ptrace but not preload. I've fixed this for preload, and added additional checks for ptrace so that we don't keep record of shared memory blocks of size 0.